contracts/GnosisSafe.sol

```diff
@@ -335,6 +334,20 @@ contract GnosisSafe
         }
         return EIP1271_MAGIC_VALUE;
     }
+    
+    /// @dev Returns the chain id used by this contract.
+    function getChainId() public view returns (uint256) {
+        uint256 id;
+        // solium-disable-next-line security/no-inline-assembly
+        assembly {
+            id := chainid()
+        }
+        return id;
+    }
+
+    function domainSeparator() public view returns (bytes32) {
+        return keccak256(abi.encode(DOMAIN_SEPARATOR_TYPEHASH, getChainId(), this));
```

Comment:
Why wasn't it implemented as a function last time? Gas?

Reply:
because the opcode did not exist yet ;)

Reply:
I mean the `domainSeparator`, not the `chainId` 😉

Reply:
if you meant the domain separator then because at that point gas storage was cheaper so it was more gas efficient to store it on creation :)